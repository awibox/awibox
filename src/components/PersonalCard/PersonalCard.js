import React from 'react';
import PropTypes from 'prop-types';
// Styles
import styles from './PersonalCard.scss';

const PersonalCard = (props) => {
  const { data } = props;
  return (
    <div className={styles.profile}>
      <div className={styles.avatar} style={{ backgroundImage: `url(${data.get('avatar')})` }}/>
      <div className={styles.profileItems}>
        <div className={styles.profileItem}>
          <div className={styles.icon}><i className='fa fa-building'/></div>
          <div className={styles.name}>{data.get('location')}</div>
        </div>
        {data.get('github') && <div className={styles.profileItem}>
          <div className={styles.icon}><i className='fa fa-github'/></div>
          <div className={styles.name}>
            <a href={data.get('github')}
               target='_blank'
               rel='noopener noreferrer'>{data.get('github').split('/').reverse()[0]}</a>
          </div>
        </div>}
        {data.get('linkedIn') && <div className={styles.profileItem}>
          <div className={styles.icon}>
            <i className='fa fa-linkedin-square' style={{ color: '#0077B5' }}/>
          </div>
          <div className={styles.name}>
            <a href={data.get('linkedIn')}
               target='_blank'
               rel='noopener noreferrer'>{data.get('linkedIn').split('/').reverse()[0]}</a>
          </div>
        </div>
        }
        <div className={styles.profileItem}>
          <div className={styles.icon}>
            <i className='fa fa-envelope-open-o'/>
          </div>
          <div className={styles.name}>
          <a href={`mailto:${data.get('email')}`}
             target='_blank'
             rel='noopener noreferrer'>{data.get('email')}</a>
          </div>
        </div>
      </div>
    </div>
  );
};

PersonalCard.propTypes = {
  data: PropTypes.any.isRequired,
};

export default PersonalCard;
