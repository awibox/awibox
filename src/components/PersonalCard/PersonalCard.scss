@import '../../styles/variables';

$avatarSize: 150px;

.profile {
  display: flex;
  flex-flow: row;
}
.avatar {
  width: $avatarSize;
  min-width: $avatarSize/2;
  height: $avatarSize;
  background-size: contain;
  background-repeat: no-repeat;
  background-position: 50% 50%;
}
.profileItems {
  display: flex;
  flex-flow: column;
  padding-top: 15px;
}
.profileItem {
  display: flex;
  flex-flow: row;
  align-items: center;
  padding: 5px 0 5px 20px;
  .icon {
    font-size: 20px;
    width: 20px;
    margin-right: 10px;
  }
  .name {
  }
  a {
    color: $blue;
  }
}

@media (max-width: $widthScreenMobileSmall) {
  .profile {
    flex-flow: column;
  }
  .avatar {
    width: 100%;
  }
}
