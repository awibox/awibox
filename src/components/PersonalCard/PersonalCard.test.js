import React from 'react';
import { shallow } from 'enzyme';
import { Map } from 'immutable';
import PersonalCard from './PersonalCard';

const testData = Map({
  avatar: 'https://avatars1.githubusercontent.com/u/1931010?s=460&v=4',
  position: 'Test position',
  location: 'Palo Alto',
  email: 'test@test.com',
  services: [
    {
      name: 'github',
      icon: 'github',
      link: 'https://github.com/',
      color: '#24292e',
    },
    {
      name: 'linkedin',
      icon: 'linkedin-square',
      link: 'https://linkedin.com/',
      color: '#0077B5',
    },
  ],
});

describe('Alert', () => {
  it('should render correctly', () => {
    const component = shallow(<PersonalCard data={testData}/>);
    expect(component).toMatchSnapshot();
  });
});
