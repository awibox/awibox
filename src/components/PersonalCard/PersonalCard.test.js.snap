// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly 1`] = `
<div
  className="profile"
>
  <div
    className="avatar"
    style={
      Object {
        "backgroundImage": "url(https://avatars1.githubusercontent.com/u/1931010?s=460&v=4)",
      }
    }
  />
  <div
    className="profileItems"
  >
    <div
      className="profileItem"
    >
      <div
        className="icon"
      >
        <i
          className="fa fa-user"
        />
      </div>
      <div>
        Test position
      </div>
    </div>
    <div
      className="profileItem"
    >
      <div
        className="icon"
      >
        <i
          className="fa fa-location-arrow"
        />
      </div>
      <div>
        Palo Alto
      </div>
    </div>
    <div
      className="profileItem"
    >
      <div
        className="icon"
      >
        <i
          className="fa fa-envelope-open-o"
        />
      </div>
      <div>
        <a
          href="mailto:test@test.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          test@test.com
        </a>
      </div>
    </div>
    <div
      className="profileServices"
    >
      <div
        className="icon"
        key="github"
      >
        <a
          href="https://github.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="fa fa-github"
            style={
              Object {
                "color": "#24292e",
              }
            }
          />
        </a>
      </div>
      <div
        className="icon"
        key="linkedin"
      >
        <a
          href="https://linkedin.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <i
            className="fa fa-linkedin-square"
            style={
              Object {
                "color": "#0077B5",
              }
            }
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
